tool_collections/galaxy_sequence_utils/fastq_manipulation/fastq_manipulation.xml

```diff
@@ -2,9 +2,9 @@
   <options sanitize="False" /> <!-- This tool uses a file to rely all parameter information (actually a dynamically generated python module), we can safely not sanitize any parameters -->
   <description>reads on various attributes</description>
   <requirements>
-    <requirement type="package" version="1.0.1">galaxy_sequence_utils</requirement>
+    <requirement type="package" version="1.1.0">galaxy_sequence_utils</requirement>
   </requirements>
-  <command>python $__tool_directory__/fastq_manipulation.py '$input_file' '$fastq_manipulation_file' '$output_file' '$output_file.files_path' '${input_file.extension[len( 'fastq' ):]}'</command>
+  <command>gx-fastq-manipulation '$input_file' '$fastq_manipulation_file' '$output_file' '$output_file.files_path' '${input_file.extension[len( 'fastq' ):]}'</command>
   <configfiles>
     <configfile name="fastq_manipulation_file">##create an importable module
 #import binascii
```

Comment:
The code in this `configfile` is not Python3 compatible (e.g. uses `string.maketrans()`). Not sure keeping this big block of code here is a good idea anyway.

Reply:
@jmchilton I've fixed the problem with `string.maketrans` in Python3, but I still don't understand what is the purpose of using `binascii.unhexlify("${ binascii.hexlify(str(match_block['match_type']['match']['match_by'])) }")` instead of just `$match_block['match_type']['match']['match_by']` for all text params, any clue?

Reply:
I don't have a clue... security related maybe?

Reply:
If somehow this circumvents the *Wrapper crap Galaxy does - this could make sense I suppose. I thought the parameters were still wrapped when using in configfiles though - I could be wrong.

Reply:
Ping @blankenberg

Reply:
@jmchilton is correct with his thinking here. The sanitizer has been turned off on this tool, so we avoid issues by hexlify user text values.

Reply:
Thanks @blankenberg! Do you think that's still the best way to implement this?

Reply:
Good question @nsoranzo. This fastq manipulation tool is very Galaxy-centric in its design. To allow the full complement of user text input values, while preventing security issues it is probably hard to beat -- properly quoting and unquoting strings safely can be difficult and this avoids any complications with ease. 

Rewriting the tool to no longer use configfiles, but instead make use of long command-line option sets may make it more generally useful outside of Galaxy, but isn't something I'd have any time to explore currently.

Reply:
@jmchilton @blankenberg I've fixed this by adding a `.decode()` after each `binascii.hexlify()`.